homeassistant/components/rfxtrx/light.py

```diff
@@ -72,14 +73,37 @@ def light_update(event):
         RECEIVED_EVT_SUBSCRIBERS.append(light_update)
 
 
-class RfxtrxLight(RfxtrxDevice, Light):
+class RfxtrxLight(RfxtrxDevice, Light, RestoreEntity):
     """Representation of a RFXtrx light."""
 
+    async def async_added_to_hass(self):
+        """Restore RFXtrx device state (ON/OFF)."""
+        await super().async_added_to_hass()
+
+        old_state = await self.async_get_last_state()
+        if old_state is not None:
+            self._state = old_state.state == STATE_ON
+
+        # Restore the brightness of dimmable devices
+        if (
+            old_state is not None
+            and old_state.attributes.get(ATTR_BRIGHTNESS) is not None
+        ):
+            self._brightness = int(old_state.attributes[ATTR_BRIGHTNESS])
+
     @property
     def brightness(self):
         """Return the brightness of this light between 0..255."""
         return self._brightness
 
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attr = {}
+        if self._brightness is not None:
+            attr[ATTR_BRIGHTNESS] = self._brightness
```

Comment:
Brightness is already part of default light state attributes. We should remove this change.

Reply:
@MartinHjelmare You're right. I can leave out the entire 'def device_state_attributes(self):' section out (line 99 - 105). Do you agree? 

I've tested it without this part and it restores the state fine. Strangely, during earlier testing, this didn't work and found #20588 where this part was added in the RFlink component. 

Can you confirm I can leave out the entire section (line 99-105)? I will make a new PR later today.

Reply:
Yes, please enoce the whole property. :+1:

Reply:
Should be resolved with #30397